### PR TITLE
Allow wrapping event processing and commit in a transaction

### LINKED ISSUE #178: Transactions around processing of a batch
**As** a client
**I want** to be able to decorate the `void processBatch(Batch<T> batch)` method
**So that** I can have a transaction around it.

See [this method](https://github.com/zalando-incubator/fahrschein/blob/master/fahrschein/src/main/java/org/zalando/fahrschein/NakadiReader.java#L195). Currently if `cursorManager.onSucess` fails, a client has no chance to roll back whatever she did in the `listener`.

### PR DESCRIPTION
**Pull request by jhorstmann**
Fixes #178

### CODE REVIEW

File: fahrschein-jdbc/src/main/java/org/zalando/fahrschein/jdbc/TransactionalBatchHandler.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package org.zalando.fahrschein.jdbc;
```
Comment (lukasniemeier-zalando) [inline fahrschein-jdbc/src/main/java/org/zalando/fahrschein/jdbc/TransactionalBatchHandler.java]: Or move to core fahrschein and have `spring-tx` as `provided` dependency?

Comment (jhorstmann) [inline fahrschein-jdbc/src/main/java/org/zalando/fahrschein/jdbc/TransactionalBatchHandler.java]: I would consider this class more as an example, you might want to customize the rollback behaviour.

**Comment by jhorstmann** [comment]
:+1:

**Comment by lukasniemeier-zalando** [comment]
:+1:

**Pull request merged**